vsphere_plugin_common/__init__.py

```diff
@@ -1465,6 +1468,48 @@ def _add_network(self, network, datacenter):
             guest_map.adapter.subnetMask = str(nw.netmask)
         return nicspec, guest_map
 
+    def _get_nic_keys_for_remove(self, server):
+        # get nics keys in tempalte before our changes,
+        # should be used keys instead mac, as macs will be changed after
+        # create
+        keys = []
+        for device in server.config.hardware.device:
+            # delete network interface
+            if hasattr(device, 'macAddress'):
+                keys.append(device.key)
+        return keys
+
+    def _remove_nic_keys(self, server, keys):
+        self._logger.debug(
+            'Removing network adapters {keys} from vm. '
+            .format(keys=repr(keys)))
+        # remove nic's by key
+        for key in keys:
+            devices = []
+            for device in server.config.hardware.device:
+                # delete network interface
+                if hasattr(device, 'macAddress') and key == device.key:
```

Comment:
Do we need to check that it has a MAC address? Isn't the key a UUID, and therefore it shouldn't be reasonably possible for there to be a collision?

Reply:
Removed